### ISSUE TITLE
Docs: layout data fetching is using `currently`, does that mean soon it can become possible? Or is this wishful thinking?

### ISSUE THREAD

**Issue by maapteh** [vercel/next.js#33061]
### What is the improvement or update you wish to see?

On https://nextjs.org/docs/basic-features/layouts#data-fetching it states: 

```
Inside your layout, you can fetch data on the client-side using useEffect or a library like SWR. 
Because this file is not a Page, you cannot use getStaticProps or getServerSideProps currently.
```

So the word `currently` is used. That can mean that can change, and if so any plans on when? I was hoping to have all my "problems" fixed in 12 :) but cant find a nice way to SSR header and footer data only once. Currently every page loads this data and then it bubbles down.

### Is there any context that might help us understand?

Im just hoping to finally be able to load layout data once and have all Next stuff working (static generating etc etc), still looking at all 12 options but couldn't find it being solved so hence trying to understand if currently can mean it will become possible.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/basic-features/layouts

**Comment by leerob** [comment]
"Currently" is in reference to React Server Components. More updates and discussion here 😄 

- https://github.com/vercel/next.js/discussions/10949

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
